spark-cassandra-connector/src/it/scala/org/apache/spark/sql/cassandra/CassandraSQLPredicatePushdownSpec.scala

```diff
@@ -50,29 +54,29 @@ class CassandraSQLPredicatePushdownSpec extends SparkCassandraITFlatSpecBase {
 
   it should "allow predicate pushdown if all partition columns have EQ predicate.  a=1 and b=1 and c=1 and d =1 are pushed down" in {
     val plan = cc.sql("select * from test_pd where a=1 and b=1 and c=1 and d=1").queryExecution.sparkPlan
-    plan.nodeName shouldBe "CassandraTableScan"
-    plan.children should have size 0
+    plan.nodeName shouldBe "Filter"
+    plan.children should have size 1
```

Comment:
Not part of this ticket, but there are tests in this file authored by you, which lack those kinds of checks. E.g. one test above this, you have a test:
```scala
 it should "not allow predicate pushdown if some partition column doesn't have predicate" in {
    val result = cc.sql("select * from test_pd where a=1 and b=1 and c=1").collect()
  }
```
I have no idea how this test validates what it promises to validate. It doesn't check if the predicate pushdown was or wasn't really applied and it doesn't even check the correctness of the result. This ticket, when you already touched those predicate pushdown tests, is a good occasion to improve those other tests without proper checks.